Storage/test.cc

```diff
@@ -283,6 +354,47 @@ TEST(TransactionalStorage, SmokeTest) {
       }).Go()));
     }
 
+    // Iterate over the One2One with deleted elements, confirm the integrity of `forward_` and `transposed_`.
+    {
+      EXPECT_FALSE(WasCommitted(storage.Transaction([](MutableFields<Storage> fields) {
+        EXPECT_TRUE(fields.o.Rows().Has(1));
+        EXPECT_TRUE(fields.o.Cols().Has("one"));
+        EXPECT_TRUE(fields.o.Cols().Has("too"));
+
+        fields.o.Add(Cell{3, "two", 5});
+        EXPECT_TRUE(fields.o.Rows().Has(3));
+        EXPECT_FALSE(fields.o.Cols().Has("too"));
+        EXPECT_TRUE(fields.o.Cols().Has("two"));
+        EXPECT_EQ(2u, fields.o.Rows().Size());
+        EXPECT_EQ(2u, fields.o.Cols().Size());
+
+        fields.o.EraseCol("two");
+        EXPECT_FALSE(fields.o.Rows().Has(3));
+        EXPECT_FALSE(fields.o.Cols().Has("too"));
+        EXPECT_FALSE(fields.o.Cols().Has("two"));
+        EXPECT_EQ(1u, fields.o.Rows().Size());
+        EXPECT_EQ(1u, fields.o.Cols().Size());
+
+        std::multiset<std::string> rows;
+        std::multiset<std::string> cols;
```

Comment:
( without looking at the context ) I recall I used `multiset` in the Matrix test before, to keep counters, as the relationship was two-to-three.